src/mlpack/tests/load_save_test.cpp

```diff
@@ -2528,6 +2528,165 @@ TEST_CASE("LoadCSVHeaderTest", "[LoadSaveTest]")
   REQUIRE(headers.at(3) == "d");
 }
 
+TEST_CASE("DataOptionsTest", "[LoadSaveTest]")
+{
+  data::DataOptions opts1, opts2, opts3;
+ 
+  opts1.Fatal() = false;
+  opts2.Fatal() = false;
+  opts1.Format() = FileType::FileTypeUnknown;
+  opts2.Format() = FileType::CSVASCII;
+  opts3 = opts1 | opts2;
+  REQUIRE(opts3.Fatal() == false);
+  REQUIRE(opts3.Format() == FileType::CSVASCII);
+
+  opts1.Fatal() = true;
+  opts2.Fatal() = false;
+  opts1.Format() = FileType::AutoDetect;
+  opts2.Format() = FileType::RawASCII;
+  opts3 = opts1 | opts2;
+  REQUIRE(opts3.Fatal() == true);
+  REQUIRE(opts3.Format() == FileType::RawASCII);
+
+  opts1.Fatal() = true;
+  opts2.Fatal() = true;
+  opts1.Format() = FileType::RawASCII;
+  opts2.Format() = FileType::AutoDetect;
+  opts3 = opts1 | opts2;
+  REQUIRE(opts3.Fatal() == true);
+  REQUIRE(opts3.Format() == FileType::RawASCII);
+
+  opts1.Fatal() = false;
+  opts2.Fatal() = true;
+  opts1.Format() = FileType::CSVASCII;
+  opts2.Format() = FileType::FileTypeUnknown;
+  opts3 = opts1 | opts2;
+  REQUIRE(opts3.Fatal() == true);
+  REQUIRE(opts3.Format() == FileType::CSVASCII);
```

Comment:
What happens if we leave the defaults unspecified?  I think we should test the case where we do things like:

```
DataOptions opts1, opts2;
opts2.Fatal() = true;

opts3 = opts1 | opts2;

REQUIRE(opts3.Fatal() == true);
// then check that the others are the defaults
```